src/Http/Controllers/Admin/ModuleController.php

```diff
@@ -1098,39 +1098,28 @@ public function show($id, $submoduleId = null)
         return $this->redirectToForm($this->getParentModuleIdFromRequest($this->request) ?? $submoduleId ?? $id);
     }
 
-    /**
-     * @param int $id
-     * @param int|null $submoduleId
-     * @return \Illuminate\Http\JsonResponse|\Illuminate\Http\RedirectResponse|\Illuminate\View\View
-     */
-    public function edit(TwillModelContract|int $id, $submoduleId = null)
+    public function edit(TwillModelContract|int $id): mixed
     {
-        $params = $this->request->route()->parameters();
-
-        $this->submodule = count($params) > 1;
-        $this->submoduleParentId = $this->submodule
-            ? $this->getParentModuleIdFromRequest($this->request) ?? $id
-            : head($params);
-
         if ($id instanceof TwillModelContract) {
             $item = $id;
             $id = $item->id;
         } else {
-            $id = last($params);
-
-            $item = $this->repository->getById($submoduleId ?? $id);
+            $parameter = Str::singular(Str::afterLast($this->moduleName, '.'));
+            $id = $this->request->route()->parameter($parameter, $id);
+            $item = $this->repository->getById($id, $this->formWith, $this->formWithCount);
```

Comment:
@antonioribeiro this is a slightly simplified version to handle the id's.

This avoids the use of last and is more strict. If there is no argument for the current module in the url, it defaults to the id being passed in. This (should) work for you case.